platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/daemon/configuration/DaemonJvmToolchainCriteriaOptions.java

```diff
@@ -0,0 +1,103 @@
+
+import java.util.List;
+
+public class DaemonJvmToolchainCriteriaOptions extends BuildOptionSet<DaemonJvmToolchainSpec> {
+
+    private static List<BuildOption<DaemonJvmToolchainSpec>> options = ImmutableList.of(
+        new ToolchainVersionOption(),
+        new ToolchainVendorOption(),
+        new ToolchainImplementationOption()
+    );
+
+    public static List<BuildOption<DaemonJvmToolchainSpec>> get() {
+        return options;
+    }
+
+    @Override
+    public List<BuildOption<DaemonJvmToolchainSpec>> getAllOptions() {
+        return options;
+    }
+
+    public static class ToolchainVersionOption extends StringBuildOption<DaemonJvmToolchainSpec> {
+
+        public ToolchainVersionOption() {
+            super(BuildPropertiesDefaults.TOOLCHAIN_VERSION_PROPERTY, PropertyOrigin.BUILD_PROPERTIES);
+        }
+
+        @Override
+        public void applyTo(String value, DaemonJvmToolchainSpec settings, Origin origin) {
+            try {
+                int integerValue = Integer.parseInt(value);
+                settings.getLanguageVersion().set(JavaLanguageVersion.of(integerValue));
+            } catch (NumberFormatException e) {
+                origin.handleInvalidValue(value, "the value should be an int");
+            } catch (IllegalArgumentException e) {
+                origin.handleInvalidValue(value, "the value should be a positive int");
+            }
+        }
+    }
+
+    public static class ToolchainVendorOption extends EnumBuildOption<JvmVendor.KnownJvmVendor, DaemonJvmToolchainSpec> {
+
+        public ToolchainVendorOption() {
+            super(BuildPropertiesDefaults.TOOLCHAIN_VENDOR_PROPERTY, JvmVendor.KnownJvmVendor.class, JvmVendor.KnownJvmVendor.values(), BuildPropertiesDefaults.TOOLCHAIN_VENDOR_PROPERTY, PropertyOrigin.BUILD_PROPERTIES);
+        }
+
+        @Override
+        public void applyTo(JvmVendor.KnownJvmVendor value, DaemonJvmToolchainSpec settings, Origin origin) {
+            assertJvmToolchainVersion(settings);
+            settings.getVendor().set(DefaultJvmVendorSpec.of(value));
+        }
+    }
+
+    public static class ToolchainImplementationOption extends EnumBuildOption<JvmImplementation, DaemonJvmToolchainSpec> {
+
+        public ToolchainImplementationOption() {
+            super(BuildPropertiesDefaults.TOOLCHAIN_IMPLEMENTATION_PROPERTY, JvmImplementation.class, JvmImplementation.values(), BuildPropertiesDefaults.TOOLCHAIN_IMPLEMENTATION_PROPERTY, PropertyOrigin.BUILD_PROPERTIES);
+        }
+
+        @Override
+        public void applyTo(JvmImplementation value, DaemonJvmToolchainSpec settings, Origin origin) {
+            assertJvmToolchainVersion(settings);
+            settings.getImplementation().set(value);
+        }
+    }
+
+    private static void assertJvmToolchainVersion(DaemonJvmToolchainSpec settings) {
+        if (!settings.getLanguageVersion().isPresent()) {
+            String exceptionMessage = String.format("Option %s undefined on Build properties. " +
```

Comment:
```suggestion
            String exceptionMessage = String.format("Option %s undefined on build properties. " +
```

Reply:
Gradle is a proper noun. It is capitalised. Build is not, and should not be capitalized.